src/sagemaker/fw_utils.py

```diff
@@ -188,6 +189,41 @@ def validate_source_code_input_against_pipeline_variables(
         )
 
 
+def parse_mp_parameters(params):
+    """Parse the model parallelism parameters provided by the user.
+
+    Args:
+        params: a string representing path to an existing config, or
+                a config dict.
+
+    Returns:
+        parsed: a dict of parsed config.
+
+    Raises:
+        ValueError: if params is not a string or a dict, or
+                    the config file cannot be parsed as json.
+    """
+    parsed = None
+    if isinstance(params, dict):
+        parsed = params
+    elif os.path.exists(params):
+        try:
+            with open(params, "r") as fp:
+                parsed = json.load(fp)
+        except json.decoder.JSONDecodeError:
+            pass
+    else:
+        raise ValueError(
+            f"Expected a string path to an existing modelparallel config, or a dictionary. "
+            f"Received: {params}."
+        )
+
+    if parsed is None:
+        raise ValueError(f"Cannot parse {params} as a json file.")
```

Comment:
nit: this should go on line 214 right, or do you want to make sure that the dict is not empty

Reply:
- If the input `params` is an empty dict `{}`, it is okay to return it directly.
- If the input `params` is None, the exception on line 216 should be triggered.

In the current implementation, it can go either the current place, or line 214. I was thinking, in the future, if we want to extend the support to other file types (e.g., yaml), the raise in the current place can summarize the file reading failure.

Reply:
ok sounds good